app/models/metric/common.rb

```diff
@@ -52,10 +52,8 @@ def v_derived_storage_used
   def min_max_v_derived_storage_used(mode)
     cond = ["resource_type = ? and resource_id = ? and capture_interval_name = 'hourly' and timestamp >= ? and timestamp < ?",
             resource_type, resource_id, timestamp.to_date.to_s, (timestamp + 1.day).to_date.to_s]
-    meth = mode == :min ? :first : :last
-    recs = MetricRollup.where(cond)
-    rec = recs.sort { |a, b| (a.v_derived_storage_used && b.v_derived_storage_used) ? (a.v_derived_storage_used <=> b.v_derived_storage_used) : (a.v_derived_storage_used ? 1 : -1) }.send(meth)
-    rec.nil? ? nil : rec.v_derived_storage_used
+    recs = MetricRollup.where(cond).where.not(:derived_storage_total => nil, :derived_storage_free => nil)
```

Comment:
since there is nothing special in `cond`, I would prefer to see multiple `where` clauses for each of the conditions.